New gravity variables to choose best direction to show tip

kus opened the pull request:
New gravity variables to choose best direction to show tip with priority towards a direction if its able to show it in both directions.

New variables: ns, sn, ew, we

kus committed:
New gravity variables to chose best direction to show tip

jaz303 commented:
How does this differ in theory from `$.fn.tipsy.autoNS` and `$.fn.tipsy.autoWE`?

kus commented:
It enables you to specify a direction you want it to display with a fallback to the opposite direction if there is not enough room.

It gets the actual height of the tooltip and the space available in the direction you want it to show and if it fits, it displays it in the direction you want, if not in the fallback direction.

The $.fn.tipsy.autoNS only checks to see if the element is in the top half or bottom half of the screen and shows the tip accordingly, though you may have an element in the top half of the screen and want the tooltip to appear above it but $.fn.tipsy.autoNS would make it appear below it.

Where as my modification will work out if there is enough space, and will display it where you want it if there is enough room.

For example:

You have a 100px high header on your website with navigation under it, and you want to display the tooltip above the navigation when a user hovers over, the tooltip will only be about 20-30px high so it will fit, so it should be able to display there, which with my modification it will.

Then say the user scrolls down the page and the navigation is at the top of the screen, the tooltip will then display below it.

kus committed:
Added CSS arrow (by MoOx https://github.com/jaz303/tipsy/pull/41)
Added CSS arrow border
Added gravity: ns, sn, we, ew
Added trigger: 'all' to trigger on hover and focus
Added class variable to dynamically add a class

kus committed:
Added CSS arrow (by MoOx https://github.com/jaz303/tipsy/pull/41)
Added CSS arrow border
Added example class "black" for different styled tooltips

kus committed:
reset to standard

kus committed:
Added CSS arrow (by MoOx https://github.com/jaz303/tipsy/pull/41)
Added CSS arrow border
Added gravity: ns, sn, we, ew
Added trigger: 'all' to trigger on hover and focus
Added class variable to dynamically add a class

kus commented:
Added a few more things:

Javascript: https://github.com/kus/tipsy/commit/01b9c2e7aa81158ab9e60733634048e84e635380
CSS: https://github.com/kus/tipsy/commit/846585228ca7f24957058c5f328225f23cb8a49e

Added CSS arrow (by MoOx https://github.com/jaz303/tipsy/pull/41)
Added CSS arrow border (you can have a border on the tip and arrow - before it would only go around the tip and looked stupid)
Added gravity: ns, sn, we, ew
Added trigger: 'all' to trigger on hover and focus (if someone tabs on to the element)
Added class variable to dynamically add a class (so you can have different styled tips, ie red for something bad, green for something good etc)

kus committed:
Added new colours for example Red, Green, Blue

kus commented:
Ive made an example page with all of the features I added.

http://blakek.us/labs/tipsy/

You can clearly see the difference between my gravity functions and your $.fn.tipsy.autoNS

Cheers,

Blake
